src/database.h

```diff
@@ -23,6 +23,9 @@ friend class Singleton<Database>;
 
 	template <class UnaryPredicate>
 	std::vector<Homebrew> Sort(UnaryPredicate pred);
+
+	template <class UnaryPredicate>
+	std::vector<Homebrew> Search(UnaryPredicate pred);
```

Comment:
I'm not super fond of this because search is basically Sort + Filter. Problem of these functions is that they don't support composability. Maybe I should refactor this into a `HomebrewListView` or whatever on which several criteria could be applied.

Can you add a FIXME comment saying that it's a hack because Sort and Filter can't be composed together?